Improve content editing features, enable preview and SEO

rccsousa opened the pull request:
Why:
- collection and content management had some issues, such as `Author` `slug` field not populating automatically (issue #13 ), social links were not being correctly pushed to the frontend;

How:
- this update fixes issue #13 by enabling author collection to generate a slug from the author name field;
- SEO plugin is now enabled, allowing for content-specific meta to be generated from the dashboard;
- Live preview and Page Preview are also now correctly enabled and available for all collections (screenshots below);
- minor changes to how Author Socials' field are rendered, making them optional, improving clarity;
- deleted unused collections and respective dependencies;
- add improvements to content editing legibility and add a toolbar to give more options to the user instead of just relying on `/` commands;

Screenshots:

## New Author socials fields
![imagem](https://github.com/user-attachments/assets/c3cd6149-d138-4f97-a2ae-b2864723b9e0)



## SEO Plugin 

Configuring meta description:
![imagem](https://github.com/user-attachments/assets/173de3f9-5f79-47c9-baf0-63f64c8c03ef)

How it is rendered when sharing:
![imagem](https://github.com/user-attachments/assets/4ccb67aa-80cb-4473-a2bf-69e3757f0551)

## Live Preview
![imagem](https://github.com/user-attachments/assets/e303550f-0017-4819-bc99-298128b5dfad)


## Updated Lexical editor

![imagem](https://github.com/user-attachments/assets/0a1a10a4-1cbc-496c-bfd7-6f05a77ed3d7)

rccsousa committed:
update: convert to clickable link to author page

rccsousa committed:
fix: change featured image container position to relative

rccsousa committed:
refactor Authors component and AuthorPills

- created pills for 2 authors and 3+ authors;
- moved Author pills to a single file located in Authors/AuthorPills;
- fixed imports

rccsousa committed:
convert single author pill to a clickable link

rccsousa committed:
make highlighted content clicable and link to content

rccsousa committed:
fix: image container was rendering even without images

rccsousa committed:
fix: image container was rendering even without images

rccsousa committed:
fix styling

rccsousa committed:
fix broken links

rccsousa committed:
make content card link hug featured image

rccsousa committed:
file renaming

rccsousa committed:
update header styling

- make header correspond to page type, changing background and font
  colors based on the type of page it is inserted instead of being
  generic

rccsousa committed:
optimize css

rccsousa committed:
update: improve header dynamic style

rccsousa committed:
update: add socials collection and include social links in footer

rccsousa committed:
fix typescript errors

rccsousa committed:
update metadata generation

- changed metadata generation, badges, favicons and open graphs,
  replacing payload boilerplate open graph and use content tailored to
  the content-sub project

rccsousa committed:
improve editing optins - meta, SEO and previewing

- enabled payload SEO plugin to automatically generate metadata for
  sharing;
- enabled live preview and increased draft saving to facilitate
  previewing;
- enabled page preview;

rccsousa committed:
update fetchers to enable draft version preview

rccsousa committed:
fix slug not generating properly upon author creation/edit

rccsousa committed:
improve clarity on author socials' fields

rccsousa committed:
re-enable author social links rendering for content pages

rccsousa committed:
fix content card styling

rccsousa committed:
remove character limit on about section

rccsousa committed:
fix: point to updated authorName field instead of name

rccsousa committed:
enable SEO and live preview

rccsousa committed:
update: add paths to existing collections

rccsousa committed:
linting

rccsousa committed:
delete unused collections

rccsousa committed:
remove link to deleted collections

rccsousa committed:
update payload config

- remove unused collections,
- temporarily disable search plugin,
- add missing collections to SEO plugin and its functions (generate
title, generate URL),
- add missing collections to Redirects plugin;

rccsousa committed:
fix SEO plugin

plugin was throwing errors upon type generation due to having
collections parameters enabled, which isn't necessary for the current
version.

rccsousa committed:
regenerate types and import map

rccsousa committed:
delete unused files and respective dependencies

rccsousa committed:
update generate meta to fetch images when using cloud storage

rccsousa committed:
update metadata generation to make use of SEO plugin

rccsousa committed:
add missing categories

and remove unused ones

rccsousa committed:
cleaning

rccsousa committed:
regenerate types

rccsousa committed:
refactor AuthorPill

- remake author pill as a standalone component;
- use conditional rendering to make it fit one, two or more
contributors;

rccsousa committed:
fix: refactor to reflect new authorpill changes

rccsousa committed:
fix conflicts and update author pill

rccsousa committed:
linting

rccsousa committed:
enforce category attribution

rccsousa committed:
remove unused categories

rccsousa committed:
enrich lexical features and improve visibility

rccsousa committed:
increase summary character limit to 1000

davelange commented on src/app/_blocks/AuthorContentGrid/index.tsx:
```diff
@@ -18,9 +18,7 @@ export default function AuthorContentGrid({content}) {
-            <div key={i} className={styles.contentCard}>
-              <ContentCard contentType={key} content={contentPiece} />
-            </div>
+            <ContentCard contentType={key} content={contentPiece} rounded={true}/>
```
When passing boolean = true props to components, you can use shorthand, just 
```
<ContentCard {...other stuff} rounded />
```

rccsousa replied:
noted!

rccsousa committed:
update heading font size enforcing

davelange commented on src/app/_components/SocialLinks/index.tsx:
```diff
@@ -1,15 +1,35 @@
-      ))}
+      )}
+
+      {socials.x.length === 1 && (
```
Why are these arrays? Can users have multiple versions of each Social profile?

rccsousa replied:
users cannot have multiple versions of each social profile, but socials are now part of a "tab" and consequently stored as an array called "Socials":

![imagem](https://github.com/user-attachments/assets/89fc7012-46fd-4361-845d-809a10f289f4)

rccsousa committed:
Merge branch 'main' of https://github.com/subvisual/content-sub into rs/improve-content-editing

rccsousa committed:
regenerate payload types

rccsousa committed:
delete unused blocks

rccsousa committed:
formating

rccsousa committed:
remake BeforeDashboard component

- remove unused seed functionality;
- add information about important features of the dashboard;
- link to quick tutorial rundown;

davelange reviewed (approved):
This is more than I can review really, but let's go
